test/e2e/models/CodeServer.ts

```diff
@@ -215,11 +217,18 @@ export class CodeServerPage {
   }
 
   /**
-   * Navigate to a code-server endpoint.  By default go to the root.
+   * Navigate to a code-server endpoint (root by default).  Then wait for the
+   * editor to become available.
    */
   async navigate(endpoint = "/") {
     const to = new URL(endpoint, await this.codeServer.address())
     await this.page.goto(to.toString(), { waitUntil: "networkidle" })
+
+    // If we aren't authenticated we'll see a login page so we can't wait until
+    // the editor is ready.
+    if (this.authenticated) {
```

Comment:
nit: Am I missing something here? "if we aren't authenticated" makes me think I'd see `if (!this.authenticated)`

Oh...I get it. I wonder if we should make it more clear

```suggestion
    // Only reload editor if authenticated. Otherwise we'll get stuck 
    // reloading the login page.
    if (this.authenticated) {
```